mountpoint-s3-crt/src/io/host_resolver.rs

```diff
@@ -43,6 +44,15 @@ impl HostResolver {
 
         Ok(Self { inner })
     }
+
+    /// Get the current number of known host addresses for a given hostname
+    pub fn get_host_address_count(&self, hostname: &AwsString, kinds: AddressKinds) -> Result<usize, Error> {
```

Comment:
It would be nice to have a test that use this with a fixed S3 endpoint and validate we get some addresses back. Thoughts?

Reply:
We don't currently have a way to actually look up a host using `HostResolver` because we don't do any resolutions ourselves (the S3 client does them), and so we haven't bound the lookup functions. I think I'd rather not add new bindings for those just to be able to write tests.

Instead, what we can probably do is write a unit test on `S3CrtClient` that validates the new metric gets emitted correctly. Writing metrics tests is hard right now, but I have a followup PR to this that updates us to the newest [`metrics` crate](https://docs.rs/metrics), which makes those tests much simpler to write. So I think we should tackle testing this as a followup.